datastore/datastore.go

```diff
@@ -282,7 +290,13 @@ func (s *CloudStore) Create(ctx context.Context, key *Key, src Entity) error {
 }
 
 func (s *CloudStore) Put(ctx context.Context, key *Key, src Entity) (*Key, error) {
-	return s.client.Put(ctx, key, src)
+	key, err := s.client.Put(ctx, key, src)
```

Comment:
I think it's probably more idiomatic to check for non nil error and early return.

Reply:
Noted.